client/components/mma/switch/SwitchReview.tsx

```diff
@@ -89,35 +135,51 @@ export const SwitchReview = () => {
 		'd MMMM',
 	);
 
-	const {
-		data: previewResponse,
-		loadingState,
-	}: { data: PreviewResponse | null; loadingState: LoadingState } =
-		useAsyncLoader(
-			() =>
-				fetch(
-					`/api/product-move/${productDetail.subscription.subscriptionId}`,
-					{
-						method: 'POST',
-						body: JSON.stringify({
-							price: newAmount,
-							preview: true,
-						}),
-						headers: {
-							'Content-Type': 'application/json',
-						},
-					},
-				),
-			JsonResponseHandler,
+	const productMoveFetch = (preview: boolean) =>
+		fetch(
+			`/api/product-move/${productDetail.subscription.subscriptionId}`,
+			{
+				method: 'POST',
+				body: JSON.stringify({
+					price: newAmount,
+					preview,
+				}),
+				headers: {
+					'Content-Type': 'application/json',
+				},
+			},
 		);
 
+	const confirmSwitch = async () => {
+		setIsSwitching(true);
+		try {
+			const response = await productMoveFetch(false);
+			const data = await response.json();
```

Comment:
could this be `await JsonResponseReader(response)` to utilise the one bit of goodness the asyncLoader pattern provides? 🤔